ProcessLib/BoundaryCondition/ConstraintDirichletBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,169 @@
+                _integration_method.getWeightedPoint(ip).getCoords(),
+                shape_matrices[ip], GlobalDim, is_axially_symmetric);
+            _detJ_times_integralMeasure.push_back(
+                shape_matrices[ip].detJ * shape_matrices[ip].integralMeasure);
+        }
+    }
+
+    /// Integration for the element with the id \c element_id.
+    /// @param x The global vector containing the values for numerical
+    /// integration.
+    /// @param t The point in time the the integration will be performed.
+    /// @param bulk_mesh The bulk mesh the process is defined on.
+    /// @param getFlux The function of the constraining process used to
+    /// calculate the flux.
+    double integrate(
+        GlobalVector const& x, double const t, MeshLib::Mesh const& bulk_mesh,
+        std::function<Eigen::Vector3d(
+            std::size_t const, MathLib::Point3d const&, double const,
+            GlobalVector const&)> const& getFlux) override
+    {
+        MathLib::Vector3 surface_element_normal;
+        if (_surface_element.getDimension() < 2)
+        {
+            auto const bulk_element_normal =
+                MeshLib::FaceRule::getSurfaceNormal(
+                    bulk_mesh.getElement(_bulk_element_id));
+            MathLib::Vector3 const edge_vector(*_surface_element.getNode(0),
+                                               *_surface_element.getNode(1));
+            surface_element_normal =
+                MathLib::crossProduct(bulk_element_normal, edge_vector);
+        } else {
+            surface_element_normal =
+                MeshLib::FaceRule::getSurfaceNormal(&_surface_element);
+        }
+
+        surface_element_normal.normalize();
+        // At the moment (2018-04-26) the surface normal is not oriented
+        // according to the right hand rule
+        // for correct results it is necessary to multiply the normal with -1
+        surface_element_normal *= -1;
+
+        std::size_t const n_integration_points =
+            _integration_method.getNumberOfPoints();
+        // integrated_value +=
+        //   int_{\Gamma_e} \alpha \cdot flux \cdot normal \d \Gamma
+        double integrated_value = 0;
+        for (std::size_t ip(0); ip < n_integration_points; ip++)
+        {
+            auto const& wp = _integration_method.getWeightedPoint(ip);
+
+            auto const bulk_element_point = MeshLib::getBulkElementPoint(
+                bulk_mesh, _bulk_element_id, _bulk_face_id, wp);
+            auto const bulk_flux =
+                getFlux(_bulk_element_id, bulk_element_point, t, x);
+
+            // TODO find solution for 2d case
+            double const bulk_grad_times_normal(
+                Eigen::Map<Eigen::RowVectorXd const>(bulk_flux.data(),
+                                                     bulk_flux.size())
+                    .dot(Eigen::Map<Eigen::RowVectorXd const>(
+                        surface_element_normal.getCoords(), 3)));
```

Comment:
Since the surface element normal does not change, it could be stored in the integration point data...

Reply:
The normal doesn't change for all integration points, am I right? For this reason, I made the normal an attribute of the object and did not move it to the integration point data.

Reply:
The current implementation is returning a constant normal vector for each element. This is not correct...